packages/context/test/acceptance/class-level-bindings.ts

```diff
@@ -177,6 +177,156 @@ describe('Context bindings - Injecting dependencies of classes', () => {
     expect(resolved.config).to.equal('test-config');
   });
 
+  it('injects an option', () => {
+    class Store {
+      constructor(
+        @inject.option('x') public optionX: number,
+        @inject.option('y') public optionY: string,
+      ) {}
+    }
+
+    ctx
+      .bind('store')
+      .toClass(Store)
+      .withOptions({x: 1, y: 'a'});
+    const store = ctx.getSync('store');
+    expect(store.optionX).to.eql(1);
+    expect(store.optionY).to.eql('a');
+  });
+
+  it('injects an option with promise value', async () => {
+    class Store {
+      constructor(@inject.option('x') public optionX: number) {}
+    }
+
+    ctx
+      .bind('store')
+      .toClass(Store)
+      .withOptions({x: Promise.resolve(1)});
+    const store = await ctx.get('store');
+    expect(store.optionX).to.eql(1);
+  });
+
+  it('injects an option when `options` is a promise', async () => {
+    class Store {
+      constructor(@inject.option('x') public optionX: number) {}
+    }
+
+    ctx
+      .bind('store')
+      .toClass(Store)
+      .withOptions(Promise.resolve({x: 1}));
+    const store = await ctx.get('store');
+    expect(store.optionX).to.eql(1);
+  });
+
+  it('injects an option when `options` is a promise', async () => {
+    class Store {
+      constructor(@inject.option('x#y') public optionY: string) {}
+    }
+
+    ctx
+      .bind('store')
+      .toClass(Store)
+      .withOptions(Promise.resolve({x: {y: 'y'}}));
+    const store = await ctx.get('store');
+    expect(store.optionY).to.eql('y');
+  });
+
+  it('injects an option with nested property', () => {
+    class Store {
+      constructor(@inject.option('x#y') public optionXY: string) {}
+    }
+
+    ctx
+      .bind('store')
+      .toClass(Store)
+      .withOptions({x: {y: 'y'}});
+    const store = ctx.getSync('store');
+    expect(store.optionXY).to.eql('y');
+  });
+
+  it("injects options if the binding key is ''", () => {
+    class Store {
+      constructor(@inject.option('') public options: object) {}
+    }
+
+    ctx
+      .bind('store')
+      .toClass(Store)
+      .withOptions({x: 1, y: 'a'});
+    const store = ctx.getSync('store');
+    expect(store.options).to.eql({x: 1, y: 'a'});
+  });
+
+  it('injects options if the binding key is a path', () => {
+    class Store {
+      constructor(@inject.option('#x') public optionX: number) {}
```

Comment:
Is my understanding correct that `@inject.option('#x')` is equivalent to `@inject.option('x')`? Extending on that example, is `@inject.option('#x.y')` equivalent to `@inject.option('x#y')`?

I find that very confusing!

Can we please pick one syntax only? I think I like most `@inject.option('#x.y')`, because it treats the options object as a high-level value bound to the context, and then uses `#` to get a deep property - this is the approach we are already familiar with.

I also suggest to rename `@inject.option` to `@inject.options`, so that `@inject.options()` returns the full options object.

```ts
// get all options
@inject.options()

// get a deep property from the options
@inject.options('#x.y')
```